share-extension/ShareViewController.m

```diff
@@ -169,11 +172,19 @@ -(void) getPostInfos {
                     if ([url isKindOfClass:[NSURL class]]){
                         postActivity.url = url;
                     } else if ([url isKindOfClass:[UIImage class]]){
-                        postActivity.fileData = UIImagePNGRepresentation((UIImage*)item);
+                        if ([itemProvider hasItemConformingToTypeIdentifier:(NSString *)kUTTypePNG]){
+                            postActivity.fileData = UIImagePNGRepresentation((UIImage*)item);
+                            postActivity.fileExtension = @"png";
+                        } else {
+                            postActivity.fileData = UIImageJPEGRepresentation((UIImage*)item, kJPEGCompressionLevel);
+                            postActivity.fileExtension = @"jpg";
+                        }
+                        
                     } else if ([url isKindOfClass:[NSData class]]){
                         postActivity.fileData = (NSData*)url;
+                        postActivity.fileExtension = [itemProvider hasItemConformingToTypeIdentifier:(NSString *)kUTTypePNG]?@"png": @"jpg";
                     }
-                    postActivity.fileExtension = @"png";
+                    [self checkForImageOrientation];
```

Comment:
The image could be sent in FileURL or NSData & the method would handle both case.